### PR TITLE
[RTM] Stop using container scopes

### PR DESCRIPTION
**Pull request by leofeyer**
Stop using container scopes but keep using the `_scope` attribute until we have agreed on another solution. Also, only register the container scope listener in Symfony 2.

### CODE REVIEW

**Comment by leofeyer** [comment]
@contao/developers Please review :)

File: src/EventListener/ScopeAwareTrait.php [hunk 1]
```diff
@@ -15,77 +15,145 @@
...
- * Provides methods to test the container scope.
+ * Provides methods to test the request scope.
  *
  * @author Andreas Schempp <https://github.com/aschempp>
+ * @author Leo Feyer <https://github.com/leofeyer>
+ *
+ * @deprecated Deprecated since Contao 4.2, to be removed in Contao 5.
+ *             Use Contao\CoreBundle\Framework\ScopeAwareTrait instead.
  */
 trait ScopeAwareTrait
```
Comment (aschempp) [inline src/EventListener/ScopeAwareTrait.php]: Why not extend the trait in framework namespace?

File: src/EventListener/ContainerScopeListener.php [hunk 2]
```diff
@@ -70,16 +73,12 @@ public function onKernelFinishRequest(FinishRequestEvent $event)
...
-        $scope = $event->getRequest()->attributes->get('_scope');
+        $request = $event->getRequest();
 
-        if (!$this->container->hasScope($scope)) {
+        if (!$request->attributes->has('_scope')) {
             return null;
         }
 
-        return $scope;
+        return $request->attributes->get('_scope');
```
Comment (Toflar) [inline src/EventListener/ContainerScopeListener.php]: This can be simplified: 

```php
return $request->attributes->get('_scope', null);
```

**Comment by leofeyer** [comment]
I have changed everything according to your suggestions. The PR is RTM for me now.

**Comment by leofeyer** [comment]
Merged in 35b6b9626c180b4741c48b305e593a5deda19d38.

**Pull request merged**